bin/smack.py

```diff
@@ -141,6 +144,9 @@ def arguments():
   verifier_group.add_argument('--loop-limit', metavar='N', default='1', type=int,
     help='upper bound on minimum loop iterations [default: %(default)s]')
 
+  verifier_group.add_argument('--context-switches', metavar='k', default='1', type=int,
```

Comment:
So there is some ambiguity for me, on this point...

Consider the following program:
```
#include <pthread.h>

int z = 1;

void *t1(void *arg) {
  pthread_mutex_t* lock = arg;
  pthread_mutex_lock(lock);
  z++;
  pthread_mutex_unlock(lock);
}

void *t2(void *arg) {
  if(z==3)
    assert(0);
}

int main() {
  pthread_mutex_t lock;
  pthread_mutex_init(&lock, 0);

  pthread_t tid1, tid2;

  pthread_create(&tid1, 0, t1, &lock);
  pthread_create(&tid2, 0, t2, &lock);
  pthread_mutex_lock(&lock);
  z++;
  pthread_mutex_unlock(&lock);
  pthread_join(tid1, 0);
  pthread_join(tid2, 0);
}
```

The only way for the assertion in `t2` to fail is if the main calling thread, and `tid1` have already incremented `z`.  With the current definition of `--context-switches`, if `--context-switches=2` (i.e., `/k:2`), it finds the bug.  However, this obviously requires considering statements in all three threads, which requires 3 contexts and 2 switches between them.

The bug is not found when `--context-switches=1`.

My original interpretation was the same as yours -- that `/k` was the number of contexts to consider, however, that isn't really the behavior that I see.

Thoughts on this?  I am fine to change this, if you want to be consistent with the way `/k` is described by Corral.  (Though, perhaps other backends may set limits in terms of context switches, rather than contexts to be considered.  Maybe it would be a good idea to take a parameter like `--context-switches`, and add one to this value before passing it to `/k` for corral...)